ast/desugar/Desugar.cc

```diff
@@ -1068,6 +1068,13 @@ ExpressionPtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what)
             [&](parser::Assign *asgn) {
                 auto lhs = node2TreeImpl(dctx, std::move(asgn->lhs));
                 auto rhs = node2TreeImpl(dctx, std::move(asgn->rhs));
+                if (isa_tree<UnresolvedConstantLit>(lhs) && isa_tree<UnresolvedConstantLit>(rhs)) {
+                    auto &rhsConst = cast_tree_nonnull<UnresolvedConstantLit>(rhs);
+                    if (rhsConst.cnst == core::Names::Constants::ErrorNode()) {
+                        auto rhsLocZero = rhs.loc().copyWithZeroLength();
+                        rhs = MK::Let(rhsLocZero, std::move(rhs), MK::Untyped(rhsLocZero));
+                    }
+                }
```

Comment:
What's going on here?

Reply:
This is code that you wrote. :sweat_smile: 

I believe what this is doing is ensuring that static field definitions with a rhs error always wind up in a form that ensures that the lhs is "properly" typed for the fast path in `typed: strict`.  I haven't really experimented with taking this part out, but I could imagine weird things happening if you did?  (OTOH, presumably we have some tests that handle malformed static fields in `typed: strict` files already...?)

Reply:
Yeah I'd be curious to see what fails when this is removed. I don't quite remember

Reply:
Ah, I see why this was required now.

In an edit path that looks like

```ruby
A = 1
A = <ErrorNode>
A = ''
```

`A` is always a static field, but sometimes it is a class alias while sometimes it is a "normal" static field.

class aliases are special in that a change to a class alias might make certain things start or stop resolving, which we currently can't check except by re-running the slow path. This `T.let` makes it so that `A = <ErrorNode>` never looks like a class alias.

This *might* bite us if/when we attempt to get all changes to constants on the fast path, but for now this seems like an improvement.

The two alternatives here would be either

```ruby
A = T.let(<ErrorNode>, T.untyped)
A = T.unsafe(<ErrorNode>)
```

the `T.let` is nice because it means there's not an extra "constants must have type annotations in `# typed: strict` files" but it's also something I'm usually reluctant to do because adding `T.let` in various places (especially inside method bodies) has an annoying tendency to cause confusing pinning errors that the user can't resolve.

Since this is a static field not a variable it's probably fine to leave it at `T.let`. If you'd like to take a stab at writing a brief comment to this effect in the file, that'd be great. If you'd prefer that I craft some sort of comment here lmk.